### PR TITLE
ISSUE-20621 Remove (optional) text from Address line 2 field.

### PR DESCRIPTION
**Pull request by vishalkakadiya**
For more details check: https://github.com/woocommerce/woocommerce/issues/20621

### CODE REVIEW

**Comment by glagonikas** [comment]
@mikejolley Shouldn't this issue be put in the 3.4.4 milestone instead?

**Comment by vishalkakadiya** [comment]
@mikejolley I have updated my PR, as per your suggestion. Let me know if need more things on it. Thanks!

File: includes/class-wc-countries.php [hunk 1]
```diff
@@ -656,12 +662,12 @@ public function get_default_address_fields() {
...
 			'address_2'  => array(
-				'placeholder'  => esc_attr__( 'Apartment, suite, unit etc. (optional)', 'woocommerce' ),
+				/* translators: use to display optional text in address2 field. */
+				'placeholder'  => sprintf( esc_attr__( 'Apartment, suite, unit etc. %s', 'woocommerce' ), esc_html( $address_2_palceholder ) ),
 				'class'        => array( 'form-row-wide', 'address-field' ),
-				'required'     => false,
 				'autocomplete' => 'address-line2',
 				'priority'     => 60,
-				'required'     => 'required' === get_option( 'woocommerce_checkout_address_2_field', 'optional' ),
+				'required'     => 'required' === $address_2_palceholder,
```
Comment (mikejolley) [inline includes/class-wc-countries.php]: This line didn't need changing.

File: includes/class-wc-countries.php [hunk 2]
```diff
@@ -656,12 +662,12 @@ public function get_default_address_fields() {
 				'priority'     => 50,
 			),
 			'address_2'  => array(
-				'placeholder'  => esc_attr__( 'Apartment, suite, unit etc. (optional)', 'woocommerce' ),
+				/* translators: use to display optional text in address2 field. */
+				'placeholder'  => sprintf( esc_attr__( 'Apartment, suite, unit etc. %s', 'woocommerce' ), esc_html( $address_2_palceholder ) ),
```
Comment (mikejolley) [inline includes/class-wc-countries.php]: Move this string into your conditional above.

File: includes/class-wc-countries.php [hunk 3]
```diff
@@ -616,6 +616,12 @@ private function trim_formatted_address_line( $line ) {
 	 * @return array
 	 */
 	public function get_default_address_fields() {
+
+		$address_2_palceholder = '';
```
Comment (mikejolley) [inline includes/class-wc-countries.php]: Typo here

File: includes/class-wc-countries.php [hunk 4]
```diff
@@ -616,6 +616,12 @@ private function trim_formatted_address_line( $line ) {
...
+		if ( 'optional' === get_option( 'woocommerce_checkout_address_2_field', 'optional' ) ) {
+			$address_2_palceholder = __( '(optional)', 'woocommerce' );
```
Comment (mikejolley) [inline includes/class-wc-countries.php]: .= to append to default string if true.

**Review by mikejolley — CHANGES_REQUESTED** [review]

File: includes/class-wc-countries.php (hunk 1 above)
Comment (vishalkakadiya) [reply includes/class-wc-countries.php]: Reverted.

File: includes/class-wc-countries.php (hunk 2 above)
Comment (vishalkakadiya) [reply includes/class-wc-countries.php]: Done.

File: includes/class-wc-countries.php (hunk 3 above)
Comment (vishalkakadiya) [reply includes/class-wc-countries.php]: Fixed

File: includes/class-wc-countries.php (hunk 4 above)
Comment (vishalkakadiya) [reply includes/class-wc-countries.php]: Done

**Comment by vishalkakadiya** [comment]
@mikejolley I have refreshed my PR, thanks!

File: includes/class-wc-countries.php [hunk 5]
```diff
@@ -616,6 +616,12 @@ private function trim_formatted_address_line( $line ) {
 	 * @return array
 	 */
 	public function get_default_address_fields() {
+
+		$address_2_placeholder = __( 'Apartment, suite, unit etc.', 'woocommerce' );
+		if ( 'optional' === get_option( 'woocommerce_checkout_address_2_field', 'optional' ) ) {
+			$address_2_placeholder .= __( ' (optional)', 'woocommerce' );
```
Comment (mikejolley) [inline includes/class-wc-countries.php]: Move this space outside of the localised string - it will make translations easier.

**Review by mikejolley — CHANGES_REQUESTED** [review]
Thanks, just one more thing to tweak and then this can merge.

File: includes/class-wc-countries.php (hunk 5 above)
Comment (vishalkakadiya) [reply includes/class-wc-countries.php]: @mikejolley Can it be something like `$address_2_placeholder .= ' ' . __( ' (optional)', 'woocommerce' );` ?

**Comment by vishalkakadiya** [comment]
@mikejolley I have refreshed my PR with that space thing, thanks!

**Comment by mikejolley** [comment]
Thanks, merged in https://github.com/woocommerce/woocommerce/pull/20639

**Pull request merged**